src/appSlice.ts

```diff
@@ -0,0 +1,73 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+export interface Todo {
+    text: string;
+    completed: boolean;
+    id: number;
+}
+
+export enum VisibilityFilter {
+    SHOW_ALL = "SHOW_ALL",
+    SHOW_COMPLETED = "SHOW_COMPLETED",
+    SHOW_ACTIVE = "SHOW_ACTIVE",
+}
+
+export interface AppState {
+    todos: Todo[];
+    visibilityFilter: VisibilityFilter;
+}
+
+let currentId = 0;
+const initialState: AppState = {
+    todos: [{ id: currentId++, text: "Use Redux", completed: false }],
+    visibilityFilter: VisibilityFilter.SHOW_ALL,
+};
+
+export const appSlice = createSlice({
+    name: "app",
+    initialState,
+    reducers: {
+        addTodo: (state, action: PayloadAction<string>) => {
+            state.todos.push({
```

Comment:
Huh?!?! Why do we mutate `state` here? 😏 

- https://redux.js.org/tutorials/fundamentals/part-8-modern-redux#immutable-updates-with-immer
- https://immerjs.github.io/immer/